### PR TITLE
Try declarative command line for optimizer and optimizer parameters in MoCo

### PR DESCRIPTION
**Pull request by aced125**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?

This PR allows one to essentially control PyTorch from the command line. Deep learning as a YAML, if you will (or at least taking a step in that direction). 

It allows one to configure the optimizer like so:

```
python run.py --optimizer AdamW --optimizer-parameters '{"lr": 1e-4, "weight_decay": 1e-3, "betas": [0.9. 0.98]}'
```

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_bolts/models/self_supervised/moco/moco2_module.py [hunk 1]
```diff
@@ -35,9 +36,15 @@ def __init__(self,
...
                  encoder_momentum: float = 0.999,
                  softmax_temperature: float = 0.07,
-                 learning_rate: float = 0.03,
-                 momentum: float = 0.9,
-                 weight_decay: float = 1e-4,
+                 # learning_rate: float = 0.03,
+                 # momentum: float = 0.9,
+                 # weight_decay: float = 1e-4,
+                 optimizer: str = "SGD",
+                 optimizer_parameters: dict = {
```
Comment (Borda) [inline pl_bolts/models/self_supervised/moco/moco2_module.py:43]: this shall be None, and assign default later in the body if needed

**Comment by Borda** [comment]
@aced125 the blocking PR was merged, can we finish this PR? 🐰

**Comment by Borda** [comment]
@ananyahjha93 how is it going here?

**Comment by Borda** [comment]
@aced125 @ananyahjha93 how is it going here? maybe @akihironitta may have a suggestion... 🐰

**Comment by akihironitta** [comment]
> mind tell us more what is the advantage of passing it as JSON, to have more freedom for other optimizers?

@aced125 I'm also curious about what's better about json...

**Comment by Borda** [comment]
@aced125 could you lease resolve conflicts so we may proceed further? :rabbit:

**Comment by Borda** [comment]
@ananyahjha93 how is it going here?
